### PR TITLE
speedreader: Make heuristics backend feature flag

### PR DESCRIPTION
**Pull request by keur**
This is more accessible for my team than a command line switch.

Resolves https://github.com/brave/brave-browser/issues/15179

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by pes10k** [comment]
@keur just to make sure I understand, can you clarify a couple of things?

1. I see two flags (Speedreeder and Speedreader Heuristics).  How do these interact?  If i want the "Heuristics" version, do i enable both?  Or only the "Speedreader Heuristics" flag?
2. I still think these names are confusing (they're both using heuristics!).  WDYT of "Speedreeder - Rules List" and "Speedreeder - Readability" or similar?

**Comment by keur** [comment]
@pes10k yeah. so the speedreader flag is the flag for the speedreader feature itself. the speedreader heuristics backend switches to the readability backend we are using. and yeah sorry about the names i'll change it.

**Comment by pes10k** [comment]
thanks @keur !

**Review by pes10k — APPROVED** [review]

**Comment by keur** [comment]
note: this already passed CI, just did a name change for some variables and they timed out. unrelated.

File: components/speedreader/features.cc [hunk 1]
```diff
@@ -11,12 +11,15 @@
...
-      "Speedreader",
+  "Speedreader",
 #if BUILDFLAG(ENABLE_SPEEDREADER_FEATURE)
       base::FEATURE_ENABLED_BY_DEFAULT
 #else
       base::FEATURE_DISABLED_BY_DEFAULT
 #endif
 };
 
+const base::Feature kSpeedreaderReadabilityBackend{
```
Comment (iefremov) [inline components/speedreader/features.cc:22]: needs a space before {

File: components/speedreader/features.cc [hunk 2]
```diff
@@ -11,12 +11,15 @@
 namespace speedreader {
 
 const base::Feature kSpeedreaderFeature {
-      "Speedreader",
+  "Speedreader",
```
Comment (iefremov) [inline components/speedreader/features.cc:14]: looks like bad spacing, needs clang-format ?

**Review by iefremov — APPROVED** [review]

File: components/speedreader/features.cc (hunk 2 above)
Comment (keur) [reply components/speedreader/features.cc:14]: clang-format also did this

File: components/speedreader/features.cc (hunk 1 above)
Comment (keur) [reply components/speedreader/features.cc:22]: clang-format did this. i agree it looks kinda wonky, but it [happens in other files too ](https://github.com/brave/brave-core/blob/676d293dcb4dd21703fd0d5b8ff134e796f1410a/components/ntp_background_images/browser/features.cc#L15-L17)

**Review by mkarolin — APPROVED** [review]
`chromium_src` changes LGTM

**Pull request merged**